src/Plugin/Commerce/Condition/OpenyMembershipsIncome.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Drupal\openy_memberships\Plugin\Commerce\Condition;
+
+use Drupal\commerce\Plugin\Commerce\Condition\ConditionBase;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\commerce_price\Price;
+
+/**
+ * Provides the income condition for orders.
+ *
+ * @CommerceCondition(
+ *   id = "openy_memberships_income",
+ *   label = @Translation("Income"),
+ *   display_label = @Translation("Income"),
+ *   category = @Translation("OpenY"),
+ *   entity_type = "commerce_order",
+ * )
+ */
+class OpenyMembershipsIncome extends ConditionBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+        'operator' => '>',
+        'amount' => NULL,
+      ] + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    $amount = $this->configuration['amount'];
+    // An #ajax bug can cause $amount to be incomplete.
+    if (isset($amount) && !isset($amount['number'], $amount['currency_code'])) {
+      $amount = NULL;
+    }
+
+    $form['operator'] = [
+      '#type' => 'select',
+      '#title' => t('Operator'),
+      '#options' => $this->getComparisonOperators(),
+      '#default_value' => $this->configuration['operator'],
+      '#required' => TRUE,
+    ];
+    $form['amount'] = [
+      '#type' => 'commerce_price',
+      '#title' => t('Amount'),
+      '#default_value' => $amount,
+      '#required' => TRUE,
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    parent::submitConfigurationForm($form, $form_state);
+
+    $values = $form_state->getValue($form['#parents']);
+    $this->configuration['operator'] = $values['operator'];
+    $this->configuration['amount'] = $values['amount'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate(EntityInterface $entity) {
+    $this->assertEntity($entity);
+    /** @var \Drupal\commerce_order\Entity\OrderInterface $order */
+    $order = $entity;
+    $income = \Drupal::routeMatch()->getParameter('income');$income=null;
```

Comment:
weird line. Can we have 2 lines? Drupal code style, please
Also, why not DI for routeMatch() ?